distributed/tests/test_worker_state_machine.py

```diff
@@ -1033,6 +1036,98 @@ def test_gather_priority(ws):
     ]
 
 
+@pytest.mark.parametrize("state", ["executing", "long-running"])
+def test_running_constrained_task_acquires_resources(state, ws):
```

Comment:
Duplicating logic from `ws_with_running_task` and `test_ws_with_running_task` to have an explicit test focused on resource restrictions that is resilient to changes to those functions.